packages/core/src/extension/extension-messenger.ts

```diff
@@ -70,37 +86,67 @@ export class ExtensionMessenger {
     }
 
     post(extensionId: string, event: ExtensionEvent): void {
-        if (!this._posters[extensionId]) {
-            const extension = this._getExtensionById(extensionId);
+        try {
+            if (!this._posters[extensionId]) {
+                const extension = this._getExtensionById(extensionId);
 
-            this._posters[extensionId] = createExtensionEventPoster<ExtensionEvent>(extension);
-        }
+                this._posters[extensionId] = createExtensionEventPoster<ExtensionEvent>(extension);
+            }
 
-        this._posters[extensionId].post(event);
+            this._posters[extensionId].post(event);
+        } catch (error) {
+            this.clearCacheById(extensionId);
+            // eslint-disable-next-line no-console
+            console.log(
+                `Unable to post event to extension(${extensionId}) because extension iframe is not mounted.\nThe event that could not be delivered:`,
+                event,
+            );
+        }
```

Comment:
I tried to build a message queue to save messages for the extension when it’s offline, but I encountered an issue.

We use an internal poster for sending internal commands like extension ready and a normal poster for broadcasting, causing messages to be delivered in random order.

In my tests, saved messages often arrive at an extension before the internal poster's "extension ready" message. As a middle-ground approach, I log the lost messages for now.